packages/flutter/lib/src/cupertino/app.dart

```diff
@@ -268,47 +269,376 @@ class _CupertinoAppState extends State<CupertinoApp> {
-            return CupertinoButton.filled(
-              child: const Icon(
-                CupertinoIcons.search,
-                size: 28.0,
-                color: CupertinoColors.white,
-              ),
-              padding: EdgeInsets.zero,
-              onPressed: onPressed,
-            );
-          },
+      child: CupertinoUserInterfaceLevel(
+        data: CupertinoUserInterfaceLevelData.base,
+        child: CupertinoTheme(
+          data: effectiveThemeData,
+          child: CupertinoSystemColors._(
+            data: CupertinoSystemColors.fromSystem ?? CupertinoSystemColors.fallbackValues,
+            child: WidgetsApp(
+              key: GlobalObjectKey(this),
+              navigatorKey: widget.navigatorKey,
+              navigatorObservers: _navigatorObservers,
+              pageRouteBuilder: <T>(RouteSettings settings, WidgetBuilder builder) =>
+                CupertinoPageRoute<T>(settings: settings, builder: builder),
+              home: widget.home,
+              routes: widget.routes,
+              initialRoute: widget.initialRoute,
+              onGenerateRoute: widget.onGenerateRoute,
+              onUnknownRoute: widget.onUnknownRoute,
+              builder: widget.builder,
+              title: widget.title,
+              onGenerateTitle: widget.onGenerateTitle,
+              textStyle: effectiveThemeData.textTheme.textStyle,
+              color: widget.color ?? CupertinoColors.activeBlue,
+              locale: widget.locale,
+              localizationsDelegates: _localizationsDelegates,
+              localeResolutionCallback: widget.localeResolutionCallback,
+              localeListResolutionCallback: widget.localeListResolutionCallback,
+              supportedLocales: widget.supportedLocales,
+              showPerformanceOverlay: widget.showPerformanceOverlay,
+              checkerboardRasterCacheImages: widget.checkerboardRasterCacheImages,
+              checkerboardOffscreenLayers: widget.checkerboardOffscreenLayers,
+              showSemanticsDebugger: widget.showSemanticsDebugger,
+              debugShowCheckedModeBanner: widget.debugShowCheckedModeBanner,
+              inspectorSelectButtonBuilder: (BuildContext context, VoidCallback onPressed) {
+                return CupertinoButton.filled(
+                  child: const Icon(
+                    CupertinoIcons.search,
+                    size: 28.0,
+                    color: CupertinoColors.white,
+                  ),
+                  padding: EdgeInsets.zero,
+                  onPressed: onPressed,
+                );
+              },
+            ),
+          ),
         ),
       ),
     );
   }
 }
+
+/// Establishes a subtree where iOS system colors resolve to the given data.
+///
+/// Typically the given [CupertinoSystemColorsData] is resolved against its own
+/// [BuildContext] using [CupertinoSystemColorsData.resolveColors].
+class CupertinoSystemColors extends InheritedWidget {
```

Comment:
This should probably go in colors.dart, netxt to CupertinoSystemColorsData.

Reply:
I assume the platform channel communication is going to happen in this file? I'd like to keep the `CupertinoSystemColors._` constructor private but I'm probably gonna need it when the engine sends me all the system colors.

Reply:
Not sure why it needs to be private; do we want to prevent developers from creating CupertinoSystemColors widgets?